packages/headless/src/features/facets/category-facet-set/category-facet-set-slice.ts

```diff
@@ -162,27 +165,29 @@ function buildCategoryFacetRequest(
 }
 
 function buildCategoryFacetValueRequest(
-  value: string
+  value: string,
+  numberOfValues: number
 ): CategoryFacetValueRequest {
   return {
     value,
-    retrieveCount: 5,
+    retrieveCount: numberOfValues,
     children: [],
     state: 'idle',
     retrieveChildren: false,
   };
 }
 
 function convertCategoryFacetValueToRequest(
-  categoryFacetValue: CategoryFacetValue
+  categoryFacetValue: CategoryFacetValue,
+  numberOfValues: number
 ): CategoryFacetValueRequest {
   const {value} = categoryFacetValue;
   return {
     value,
     state: 'selected',
     children: [],
     retrieveChildren: true,
-    retrieveCount: 5,
+    retrieveCount: numberOfValues,
```

Comment:
Based on this change and line 173, I believe it would be clearer if we called this parameter in the action `retrieveCount` instead of `numberOfValues`.